Engine/Component/ComponentCollider.cpp

```diff
@@ -0,0 +1,391 @@
+
+		int numContacts = contactManifold->getNumContacts();
+		for (int j = 0; j < numContacts; j++)
+		{
+			btManifoldPoint pt = contactManifold->getContactPoint(j);
+			if (obA->getWorldArrayIndex() == body->getWorldArrayIndex() || obB->getWorldArrayIndex() == body->getWorldArrayIndex())
+			{
+				if (pt.getDistance() < 0.0f)
+				{
+					return true;
+				}
+			}
+		}
+	}
+	return false;
+}
+
+bool ComponentCollider::DetectCollisionWith(ComponentCollider * collider)
+{
+	btVector3 body_minim;
+	btVector3 body_maxim;
+	btVector3 collider_minim;
+	btVector3 collider_maxim;
+
+	if (disable_physics && collider->disable_physics) 
+	{	
+		App->physics->world->updateAabbs();
+		body->getAabb(body_minim, body_maxim);
+		collider->body->getAabb(collider_minim, collider_maxim);
+		
+		if (!(body_maxim.getX() < collider_minim.getX() || collider_maxim.getX() < body_minim.getX()))
+		{
+			if (!(body_maxim.getY() < collider_minim.getY() || collider_maxim.getY() < body_minim.getY())) 
+			{
+				if (!(body_maxim.getZ() < collider_minim.getZ() || collider_maxim.getZ() < body_minim.getZ())) 
+				{
+					return true;
+				}
+			}
+		}
+	}
+
+	int numManifolds = App->physics->world->getDispatcher()->getNumManifolds();
+	for (int i = 0; i < numManifolds; i++)
+	{
+		btPersistentManifold* contactManifold = App->physics->world->getDispatcher()->getManifoldByIndexInternal(i);
+		const btCollisionObject* obA = contactManifold->getBody0();
+		const btCollisionObject* obB = contactManifold->getBody1();
+
+		int numContacts = contactManifold->getNumContacts();
+		for (int j = 0; j < numContacts; j++)
+		{
+			btManifoldPoint pt = contactManifold->getContactPoint(j);
+			if ((obA->getWorldArrayIndex() == body->getWorldArrayIndex() && obB->getWorldArrayIndex() == collider->body->getWorldArrayIndex()) || (obB->getWorldArrayIndex() == body->getWorldArrayIndex() && obA->getWorldArrayIndex() == collider->body->getWorldArrayIndex()))
+			{
+				if (pt.getDistance() < 0.0f)
+				{
+					return true;
+				}
+			}
+		}
+	}
+	return false;
+}
+
+void ComponentCollider::SetStatic()
+{
+	int flags = body->getCollisionFlags();
+	if (is_static)
+	{
+		flags |= body->CF_KINEMATIC_OBJECT;
+		body->setCollisionFlags(flags);
+		mass = 0.0F;
+	}
+	else
+	{
+		flags -= body->CF_KINEMATIC_OBJECT;
+		body->setCollisionFlags(flags);
+		mass = 1.0F;
+	}
+	
+}
+
+void ComponentCollider::SetRotationAxis()
+{
+	body->setAngularFactor(btVector3(int(x_axis), int(y_axis), int(z_axis)));
+}
+
+void ComponentCollider::AddForce(float3& force)
+{
+	body->applyCentralForce(btVector3(force.x, force.y, force.z));
+}
+
+void ComponentCollider::DisablePhysics(bool disable)
+{
+	disable_physics = disable;
+	DisablePhysics();
+}
+
+void ComponentCollider::DisablePhysics()
+{
+	if (disable_physics || !active)
+	{
+		body->forceActivationState(DISABLE_SIMULATION);
+	}
+	else
+	{
+		body->forceActivationState(ACTIVE_TAG);
+		/*body->forceActivationState(DISABLE_DEACTIVATION);*/
+	}
+}
```

Comment:
Please we need descriptive names for these kind of functions or I will have a mental breakdown XD